bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -173,12 +202,16 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
                     if (Try {
                       val futDone = Await.ready(fut, 30.seconds)
```

Comment:
I'm still unclear why are we mixing the Java executors with Scala futures, could we just use the Scala Future, doing the updates on a for each of the future? Or do you have to wait for completion for another reason?